### DISCUSSION TITLE
Questions on max # devices

### DISCUSSION THREAD

**Discussion by chrissv** [Koenkk/zigbee2mqtt#16402] [General]
First a pre-question: when a device first joins a network, it may connect directly with the coordinator, or with one of the connected routers. After it initially joins the network, does it always have to go through the router (or direct to the coordinator) that it used when it first joined? Or will it pick the "best" route, even through a different router?

Now the main question: I understand there is a limit to the number of devices which a coordinator can handle.
Q1: Is that a limit of paired devices, or connected (active) devices?
Q2: If the limit is "N" and there are "N" devices connected to the coordinator, and another device wants to connect, will it automatically fall back to using a different router?

I guess my main issue is if I have multiple routers in my network, will I have problems supporting a larger number of devices (greater than the per-coordinator limit).

Thanks!

-- Steven

**Comment by ahd71** [comment]
Typically all non battery devices acts as routers too. A device can connect to either controller (where there is a max limit) or to a router. It's a mesh and from most aspects act transparently. 
The exact number of nodes depends on things like firmware, settings and number of routers. 
I have a bit over 200+ devices and it works great especially with the latest firmware that addressed some issues for large networks.
In the manual pages there are some formulas to calculate size limits.

**Reply by chrissv** [reply]
> I have a bit over 200+ devices

What coordinator do you use? With which firmware?

Thanks.

**Reply by ahd71** [reply]
Zzh stick (CC2652R) with firmware from december 2022. Z2m 1.30.0

**Reply by ahd71** [reply]
https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator

Depending on firmware and hardware different number of total devices possible. See this article for example for a few popular ones.

**Reply by ahd71** [reply]
Or this one https://www.zigbee2mqtt.io/guide/faq/#i-read-that-zigbee2mqtt-has-a-limit-of-20-devices-when-using-a-cc2530-cc2531-adapter-is-this-true
